pupil-api/src/services/redis-pupil-authentication.service.ts

```diff
@@ -0,0 +1,25 @@
+import Redis, { RedisOptions } from 'ioredis'
+import config from '../config'
+import { IRedisService, RedisService } from './redis.service'
+
+export interface IPupilAuthenticationService {
+  authenticate (schoolPin: string, pupilPin: number): Promise<object>
+}
+
+export class RedisPupilAuthenticationService implements IPupilAuthenticationService {
+
+  private redisService: IRedisService
+
+  constructor (redisService: IRedisService) {
+    if (redisService === undefined) {
+      redisService = new RedisService()
+    }
+    this.redisService = redisService
+  }
+
+  async authenticate (schoolPin: string, pupilPin: number): Promise<object> {
+    await this.redisService.get(`preparedCheck:${schoolPin}:${pupilPin}`)
```

Comment:
This appears to be a straightforward reimplementation of the table storage auth.

So, does the `redisService.get` method throw if the key isn't found?

What if the current time is outside of the valid period between (`pinValidFrom`, `pinExpiresAt`)?

We are hardcoding the key format here.  Would it be better to encapsulate this is a function?

Reply:
this method is at the point of 'make the test pass after failing first time'.
should hopefully start work on the implementation today